gz.msgs not working on humble - demos broken

Hello, I'm just starting with ROS2 and the new gazebo, so I have been trying the demos, but they don't seem to work (which makes all quite a bit confusing for a new user of both things). I managed to make them work, the issue seems to be the use of gz.msgs on Humble/Fortress/IgnitionGazebo6.

## Environment
* OS Version: Ubuntu 22.04 
* Source or binary build?
  Source, humble branch
* Ignition Gazebo 6, installed via 'apt install ros-humble-ros-gz-sim', which should be same you get using ros-humble-desktop-full

## Description
* Expected behavior: Demos work and demonstrate
* Actual behavior: Demos fail as all messages in gz.msgs are not found, thus bridging fails

## Steps to reproduce
Launch 'ros2 launch ros_gz_sim_demos tf_bridge.launch.py'

## Output
<!-- Provide screenshots, console logs, backtraces, and/or anything that could
be useful to us resolving this issue -->
```
ros2 launch ros_gz_sim_demos tf_bridge.launch.py 
[INFO] [launch]: All log files can be found below /home/april/.ros/log/2022-12-20-18-02-04-051812-alien17r3-2204-new-3537012
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [ign-1]: process started with pid [3537025]
[INFO] [parameter_bridge-2]: process started with pid [3537027]
[INFO] [rviz2-3]: process started with pid [3537029]
[parameter_bridge-2] [INFO] [1671555724.341193953] [ros_gz_bridge]: Creating GZ->ROS Bridge: [/world/default/model/double_pendulum_with_base0/joint_state (gz.msgs.Model) -> /world/default/model/double_pendulum_with_base0/joint_state (sensor_msgs/msg/JointState)] (Lazy 0)
[parameter_bridge-2] [WARN] [1671555724.341501747] [ros_gz_bridge]: Failed to create a bridge for topic [/world/default/model/double_pendulum_with_base0/joint_state] with ROS2 type [/world/default/model/double_pendulum_with_base0/joint_state] to topic [sensor_msgs/msg/JointState] with Gazebo Transport type [gz.msgs.Model]
[parameter_bridge-2] [INFO] [1671555724.341534787] [ros_gz_bridge]: Creating GZ->ROS Bridge: [/model/double_pendulum_with_base0/pose (gz.msgs.Pose_V) -> /model/double_pendulum_with_base0/pose (tf2_msgs/msg/TFMessage)] (Lazy 0)
[parameter_bridge-2] [WARN] [1671555724.341593543] [ros_gz_bridge]: Failed to create a bridge for topic [/model/double_pendulum_with_base0/pose] with ROS2 type [/model/double_pendulum_with_base0/pose] to topic [tf2_msgs/msg/TFMessage] with Gazebo Transport type [gz.msgs.Pose_V]
[rviz2-3] [INFO] [1671555724.717600684] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-3] [INFO] [1671555724.717704446] [rviz2]: OpenGl version: 4.6 (GLSL 4.6)
[rviz2-3] [INFO] [1671555724.786843817] [rviz2]: Stereo is NOT SUPPORTED
[ign-1] libEGL warning: DRI2: failed to create dri screen
[ign-1] libEGL warning: DRI2: failed to create dri screen
^C[WARNING] [launch]: user interrupted with ctrl-c (SIGINT)
[rviz2-3] [INFO] [1671555753.971278015] [rclcpp]: signal_handler(signum=2)
[parameter_bridge-2] [INFO] [1671555753.971277617] [rclcpp]: signal_handler(signum=2)
[INFO] [rviz2-3]: process has finished cleanly [pid 3537029]
[ERROR] [parameter_bridge-2]: process[parameter_bridge-2] failed to terminate '5' seconds after receiving 'SIGINT', escalating to 'SIGTERM'
[ERROR] [ign-1]: process[ign-1] failed to terminate '5' seconds after receiving 'SIGINT', escalating to 'SIGTERM'
[INFO] [parameter_bridge-2]: sending signal 'SIGTERM' to process[parameter_bridge-2]
[INFO] [ign-1]: sending signal 'SIGTERM' to process[ign-1]
[INFO] [parameter_bridge-2]: process has finished cleanly [pid 3537027]
[ERROR] [ign-1]: process has died [pid 3537025, exit code -15, cmd 'ign gazebo -r /home/april/ros2_humble/install/ros_gz_sim_demos/share/ros_gz_sim_demos/models/double_pendulum_model.sdf'].
```
## Solution:
Either roll back to use ignition.msgs (see [fork](https://github.com/c-rizz/ros_gz/commit/50b7859de1b572b45d830f093c97fe1c2dcdf526)) or somehow make gz.msgs work.

--- a/ros_gz_bridge/resource/get_mappings.cpp.em
+++ b/ros_gz_bridge/resource/get_mappings.cpp.em
@@ -32,7 +32,7 @@ get_gz_to_ros_mapping(const std::string & gz_type_name, std::string & ros_type_n
 @[end if]@
 
 @[for m in mappings]@
-  if (gz_type_name == "@(m.gz_string())")
+  if (gz_type_name == "@(m.gz_string())" || gz_type_name == "@(m.ign_string())")
   {
     ros_type_name = "@(m.ros2_string())";
     return true;
@@ -68,7 +68,11 @@ get_all_message_mappings_ros_to_gz()
 @[for m in mappings]@
     {
       "@(m.ros2_string())",  // ROS 2
-      "@(m.gz_string())", // Gazebo 
+      "@(m.gz_string())", // Gazebo
+    },
+    {
+      "@(m.ros2_string())",  // ROS 2
+      "@(m.ign_string())", // Gazebo
     },
 @[end for]@
   };

--- a/ros_gz_bridge/resource/pkg_factories.cpp.em
+++ b/ros_gz_bridge/resource/pkg_factories.cpp.em
@@ -39,7 +39,7 @@ get_factory__@(ros2_package_name)(
     return std::make_shared<
       Factory<
         @(m.ros2_type()),
-        @(m.gz_type())
+        @(m.ign_type())
       >
     >("@(m.ros2_string())", "@(m.gz_string())");
   }
@@ -52,10 +52,10 @@ template<>
 void
 Factory<
   @(m.ros2_type()),
-  @(m.gz_type())
+  @(m.ign_type())
 >::convert_ros_to_gz(
   const @(m.ros2_type()) & ros_msg,
-  @(m.gz_type()) & gz_msg)
+  @(m.ign_type()) & gz_msg)
 {
   ros_gz_bridge::convert_ros_to_gz(ros_msg, gz_msg);
 }
@@ -64,9 +64,9 @@ template<>
 void
 Factory<
   @(m.ros2_type()),
-  @(m.gz_type())
+  @(m.ign_type())
 >::convert_gz_to_ros(
-  const @(m.gz_type()) & gz_msg,
+  const @(m.ign_type()) & gz_msg,
   @(m.ros2_type()) & ros_msg)
 {
   ros_gz_bridge::convert_gz_to_ros(gz_msg, ros_msg);

--- a/ros_gz_bridge/resource/pkg_factories.cpp.em
+++ b/ros_gz_bridge/resource/pkg_factories.cpp.em
@@ -34,7 +34,7 @@ get_factory__@(ros2_package_name)(
 {
 @[for m in mappings]@
   if ((ros_type_name == "@(m.ros2_string())" || ros_type_name.empty()) &&
-    gz_type_name == "@(m.gz_string())")
+      (gz_type_name == "@(m.gz_string())" || gz_type_name == "@(m.ign_string())"))
   {
     return std::make_shared<
       Factory<

--- a/ros_gz_bridge/resource/pkg_factories.hpp.em
+++ b/ros_gz_bridge/resource/pkg_factories.hpp.em
@@ -29,7 +29,7 @@ namespace ros_gz_bridge
 
 std::shared_ptr<FactoryInterface>
 get_factory__@(ros2_package_name)(
-  const std::string & ros_type_name, 
+  const std::string & ros_type_name,
   const std::string & gz_type_name);
 
 }  // namespace ros_gz_bridge

--- a/ros_gz_bridge/ros_gz_bridge/__init__.py
+++ b/ros_gz_bridge/ros_gz_bridge/__init__.py
@@ -36,13 +36,21 @@ class MessageMapping:
         # Return ROS2 type of a message (eg std_msgs::msg::Bool)
         return f'{self.ros2_package_name}::msg::{self.ros2_message_name}'
 
-    def gz_string(self):
+    def ign_string(self):
         # Return GZ string version of a message (eg ignition.msgs.Bool)
         return f'ignition.msgs.{self.gz_message_name}'
 
-    def gz_type(self):
+    def ign_type(self):
         # Return GZ type of a message (eg ignition::msgs::Bool)
         return f'ignition::msgs::{self.gz_message_name}'
+
+    def gz_string(self):
+        # Return GZ string version of a message (eg ignition.msgs.Bool)
+        return f'gz.msgs.{self.gz_message_name}'
+
+    def gz_type(self):
+        # Return GZ type of a message (eg ignition::msgs::Bool)
+        return f'gz::msgs::{self.gz_message_name}'
 
     def unique(self):
         return f'{self.gz_message_name.lower()}_{self.ros2_message_name.lower()}'

--- a/ros_gz_bridge/test/resource/gz_publisher.cpp.em
+++ b/ros_gz_bridge/test/resource/gz_publisher.cpp.em
@@ -55,8 +55,8 @@ int main(int /*argc*/, char **/*argv*/)
 @[for m in mappings]@
   // @(m.gz_string()).
   auto @(m.unique())_pub =
-    node.Advertise<@(m.gz_type())>("@(m.unique())");
-  @(m.gz_type()) @(m.unique())_msg;
+    node.Advertise<@(m.ign_type())>("@(m.unique())");
+  @(m.ign_type()) @(m.unique())_msg;
   ros_gz_bridge::testing::createTestMsg(@(m.unique())_msg);
 
 @[end for]@

--- a/ros_gz_bridge/test/resource/gz_subscriber.cpp.em
+++ b/ros_gz_bridge/test/resource/gz_subscriber.cpp.em
@@ -54,7 +54,7 @@ private: ignition::transport::Node node;
 /////////////////////////////////////////////////
 TEST(GzSubscriberTest, @(m.unique()))
 {
-  MyTestClass<@(m.gz_type())> client("@(m.unique())");
+  MyTestClass<@(m.ign_type())> client("@(m.unique())");
 
   using namespace std::chrono_literals;
   ros_gz_bridge::testing::waitUntilBoolVar(

